pkg/virt-operator/kubevirt_test.go

```diff
@@ -3358,9 +3359,39 @@ var _ = Describe("KubeVirt Operator", func() {
 
 				return true, create.GetObject(), nil
 			})
+			kvTestData.kubeClient.Fake.PrependReactor("delete-collection", "configmaps", func(action testing.Action) (handled bool, obj runtime.Object, err error) {
+				return true, nil, nil
+			})
 
 			// This generates and posts the install strategy config map
-			install.DumpInstallStrategyToConfigMap(kvTestData.virtClient, NAMESPACE)
+			err = install.DumpInstallStrategyToConfigMap(kvTestData.virtClient, NAMESPACE)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should delete old install strategy on upgrade", func() {
+			kvTestData := KubeVirtTestData{}
+			kvTestData.BeforeTest()
+			defer kvTestData.AfterTest()
+
+			config, err := util.GetConfigFromEnv()
+			Expect(err).ToNot(HaveOccurred())
+
+			kvTestData.kubeClient.Fake.PrependReactor("delete-collection", "configmaps", func(action testing.Action) (handled bool, obj runtime.Object, err error) {
+				deleteCollectionAction, ok := action.(testing.DeleteCollectionAction)
+				Expect(ok).To(BeTrue())
+
+				listRestrictions := deleteCollectionAction.GetListRestrictions()
+				Expect(listRestrictions.Fields).To(BeEmpty())
+				Expect(listRestrictions.Labels.Matches(labels.Set(map[string]string{v1.InstallStrategyLabel: ""}))).To(BeTrue())
+
+				configMapNamespace := deleteCollectionAction.GetNamespace()
+				Expect(configMapNamespace).To(Equal(config.Namespace))
+
+				return true, nil, nil
+			})
```

Comment:
This might actually not be called. The test should assert this is called.

Reply:
Done :)